### PR TITLE
Include paths in output TSV

### PR DESCRIPTION
**Pull request by sjspielman**
Towards #17 

This PR updates the script and dockerfile (with dependencies) to include the file path in the returned table. I think actually in the opening issue comments @jashapiro and I had the same strategy in mind, just that his was written clearly 😄 , so that's the strategy I took! Seems to be working fine when running locally on my end. 

After this goes in, I'll do a `0.3` (and `0`) release, and close out the issue.

### CODE REVIEW

File: spell-check.R [hunk 1]
```diff
@@ -31,11 +31,27 @@ if (file.exists(dict_file)) {
...
-# Run spell check
-spelling_errors <- spelling::spell_check_files(files, ignore = dictionary) |>
-  data.frame() |>
-  tidyr::unnest(cols = found) |>
-  tidyr::separate(found, into = c("file", "lines"), sep = ":")
+
+# Separate files into path groups
+all_paths <- files |>
+  dirname() |>
+  unique()
```
Comment (jashapiro) [inline spell-check.R]: Doing a second search for files later seems like a potential problem, so why not preserve these as groups:
```suggestion
file_list <- split(files, dirname(files))
```

Then you can use purrr::imap():
```
file_list |> purrr::imap(\(files, path){
  spelling::spell_check_files(files, ignore = dictionary) |>
 ...
```

**Comment by sjspielman** [comment]
Confirming the code works!

File: spell-check.R [hunk 2]
```diff
@@ -31,11 +31,24 @@ if (file.exists(dict_file)) {
...
+file_list <- split(files, dirname(files))
+
+
+# check spelling for all files in each path, and prepend the file path
+#  to the file name in the final data frame
+spelling_errors <- file_list |>
+  purrr::imap(
+    \(files, path) {
+      spelling::spell_check_files(files, ignore = dictionary) |>
+      data.frame() |>
```
Comment (jashapiro) [inline spell-check.R]: Indentation consistency
```suggestion
        data.frame() |>
```

File: Dockerfile [hunk 3]
```diff
@@ -6,8 +6,8 @@ FROM rocker/r-ver:4.4.0
 LABEL org.opencontainers.image.authors="CCDL ccdl@alexslemonade.org"
 LABEL org.opencontainers.image.source="https://github.com/AlexsLemonade/spellcheck/tree/main"
 
-# install the spelling and tidyr packages from CRAN
-RUN Rscript -e "install.packages(c('readr', 'spelling', 'tidyr'))"
+# install R package dependencies from CRAN
+RUN Rscript -e "install.packages(c('spelling', 'readr', 'purrr', 'dplyr', 'tidyr'))"
```
Comment (jashapiro) [inline Dockerfile]: alphabetical imports
```suggestion
RUN Rscript -e "install.packages(c('dplyr', 'purrr', 'readr', 'spelling', 'tidyr'))"
```

**Review by jashapiro — APPROVED** [review]
LGTM

**Pull request merged**